src/main/java/hudson/tasks/junit/TestResult.java

```diff
@@ -306,14 +310,20 @@ public void parse(File reportFile) throws IOException {
         } catch (RuntimeException e) {
             throw new IOException("Failed to read "+reportFile,e);
         } catch (DocumentException e) {
+            String reportFileContent = null;
+            try {
+                reportFileContent = Files.toString(reportFile, UTF_8);
+            } catch (Exception ee) {
+                /* No need to further explain the error if we end up here. */
+            }
             if (!reportFile.getPath().endsWith(".xml")) {
-                throw new IOException("Failed to read "+reportFile+"\n"+
-                    "Is this really a JUnit report file? Your configuration must be matching too many files",e);
+                throw new IOException("Failed to parse "+reportFile+"\n"+
+                    "Is this really a JUnit report file? Your configuration may be matching too many files? The file contains:\n\n" + reportFileContent + "\n\n",e);
```

Comment:
We reject it because it cannot be parsed *and suggest* that it may not be a report file because of its name.

I would argue that this rarely happens. Probably happens during first run of a newly configured job... I don't see a problem with having a very large log in those rare cases.